src/deeprank_gnn/predict.py

```diff
@@ -295,53 +323,101 @@ def parse_output(csv_output: str, workspace_path: Path, chain_ids: list) -> None
     log.info(f"Output written to {csv_output}")
 
 
+def split_input_pdb(pdb_file: Path) -> list[Path]:
+    """Saves PDB models in a split folder, handling both single and multiple models."""
+    parser = PDBParser(QUIET=True)
+    structure = parser.get_structure("structure", pdb_file)
+
+    output_dir = pdb_file.parent / f"{pdb_file.stem}_split"
+    output_dir.mkdir(exist_ok=True)
+
+    saved_files = []
+    io = PDBIO()
+
+    # Check if the PDB contains multiple models
+    is_ensemble = len(list(structure)) > 1
+
+    for model in structure:
+        model_id = model.id if is_ensemble else 0  # Use 0 for single structure
+        output_file = output_dir / f"{pdb_file.stem}_model_{model_id}.pdb"
+        io.set_structure(model)
+        io.save(str(output_file))
+        saved_files.append(Path(output_file))
+
+    log.info(f"Processed {len(saved_files)} model(s) from {pdb_file.name}")
+    return saved_files
+
+
 def main():
     """Main function."""
 
     parser = argparse.ArgumentParser()
     parser.add_argument("pdb_file", help="Path to the PDB file.")
     parser.add_argument("chain_id_1", help="First chain ID.")
     parser.add_argument("chain_id_2", help="Second chain ID.")
-    parser.add_argument("model_path", help="Path to the pretrained model.")
+    parser.add_argument(
+        "num_cores", type=int, help="Number of cores to use"
+    )  # Changed to int
     args = parser.parse_args()
 
     pdb_file = args.pdb_file
     chain_id_1 = args.chain_id_1
     chain_id_2 = args.chain_id_2
-    model_path = args.model_path
-    
+    num_cores = args.num_cores
 
     identificator = Path(pdb_file).stem + f"-gnn_esm_pred_{chain_id_1}_{chain_id_2}"
     workspace_path = setup_workspace(identificator)
 
     # Copy PDB file to workspace
     src = Path(pdb_file)
-    dst = Path(workspace_path) / Path(pdb_file).name
-    shutil.copy(src, dst)
-    pdb_file = dst
+    copied_pdb_file = Path(workspace_path) / Path(pdb_file).name  # Renamed variable
+    shutil.copy(src, copied_pdb_file)
 
-    ## renumber PDB
-    renumber_pdb(pdb_file_path=pdb_file, chain_ids=[chain_id_1, chain_id_2])
+    pdb_files = split_input_pdb(copied_pdb_file)
+    num_cores = min(num_cores, MAX_cores, len(pdb_files))
+    log.info(f"Using {num_cores} cores for processing")
```

Comment:
Since you take the min here I'd just change the message to say something like "Using only N cores from num_cores: N to process N pdbs" - just to be clear but it's also good like this